third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -521,10 +521,26 @@ impl<'env> Generator<'env> {
                 self.gen_op_call(targets, id, BytecodeOperation::Pack(*mid, *sid, inst), args)
             },
             Operation::Select(mid, sid, fid) => {
-                let inst = self.env().get_node_instantiation(id);
                 let target = self.require_unary_target(id, targets);
                 let arg = self.require_unary_arg(id, args);
-                self.gen_select(target, id, mid.qualified_inst(*sid, inst), *fid, &arg)
+                // Get the instantiation of the struct. It's not contained in the select
+                // expression but in the type of it's operand.
+                if let Some((_, inst)) = self
```

Comment:
I tried to extract it into a function but it is not entirely possible because of lifetime issues. `env().get_node_type()` returns a temporary type on the stack from which you can't borrow inside of a helper function. However, I introduced a helper for `get_node_type` reducing a little of the boilerplate, and applied the new helper over the code.

Reply:
Yeah, I now understand this pain of Rust.  (Lifetimes block abstraction.)